flink-core/src/main/java/org/apache/flink/configuration/ClusterOptions.java

```diff
@@ -205,6 +205,24 @@ public class ClusterOptions {
                                             TextElement.code(PROCESS_WORKING_DIR_BASE.key()))
                                     .build());
 
+    public static final ConfigOption<Long> CLIENT_ALIVENESS_CHECK_INTERVAL =
+            ConfigOptions.key("cluster.client-aliveness-check.interval")
+                    .longType()
+                    .defaultValue(60000L)
+                    .withDescription(
+                            Description.builder()
+                                    .text(
+                                            "The interval to check whether the client's "
+                                                    + "heartbeat is timeout when '%s' and '%s' are "
+                                                    + "both true. Cancel the job if timeout. The "
+                                                    + "client's heartbeat interval and timeout are "
+                                                    + "set by 'client.heartbeat.interval' and "
+                                                    + "'client.heartbeat.timeout'.",
+                                            TextElement.text(DeploymentOptions.ATTACHED.key()),
+                                            TextElement.text(
+                                                    DeploymentOptions.SHUTDOWN_IF_ATTACHED.key()))
+                                    .build());
+
```

Comment:
I'm not sure if we need a dedicated config option for the checking interval. The checking interval don't need to be very accurate and therefore we probably can just use the configured value of the client heartbeat interval on the cluster side.